gslib/tests/util.py

```diff
@@ -189,10 +217,31 @@ def _HasS3Credentials():
 def _HasGSHost():
   return boto.config.get('Credentials', 'gs_host', None) is not None
 
+
+def string_compare(a, b, encoding='utf-8'):
```

Comment:
We really only use this in one place, just a few lines below.  And I feel like this could just be avoided if we coerced (via `six.ensure_foo`) the two things to be the same type before comparing them.

Reply:
There are actually a LOT of these kind of methods in this block. Removed all one-time-use, single-line methods.